core/src/main/scala/org/apache/spark/ui/scope/RDDOperationGraph.scala

```diff
@@ -258,6 +258,7 @@ private[spark] object RDDOperationGraph extends Logging {
       case DeterministicLevel.DETERMINATE => ""
       case DeterministicLevel.INDETERMINATE => " [Indeterminate]"
       case DeterministicLevel.UNORDERED => " [Unordered]"
+      case _ => ""
```

Comment:
nit: We should have added a comment on why this is required - someone looking at the current codebase will not be able to understand how this is possible :-)

Reply:
Yes, you're right. Unfortunately, this happens during loading old data generated by old Spark.

Could you make a follow-up PR, @TQJADE ?